### PR TITLE
Allow 64 bit codes and add nexa remote support.

### PR DESCRIPTION
**Pull request by abmantis**
## Description:
This adds support for rf codes with 64 bits and adds a new protocol for Nexa remotes.

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

### CODE REVIEW

File: esphome/components/remote_base/rc_switch_protocol.cpp [hunk 1]
```diff
@@ -234,13 +235,13 @@ bool RCSwitchRawReceiver::matches(RemoteReceiveData src) {
   return decoded_nbits == this->nbits_ && (decoded_code & this->mask_) == (this->code_ & this->mask_);
 }
 bool RCSwitchDumper::dump(RemoteReceiveData src) {
-  for (uint8_t i = 1; i <= 7; i++) {
+  for (uint8_t i = 1; i <= 8; i++) {
     src.reset();
     uint32_t out_data;
```
Comment (OttoWinter) [inline esphome/components/remote_base/rc_switch_protocol.cpp]: The change to 64-bit will not work - for data, `uint32_t` is used everywhere, so only 32 bits of data are supported.

You'd need to also change all references of uint32_t to uint64_t

Comment (abmantis) [reply esphome/components/remote_base/rc_switch_protocol.cpp]: @OttoWinter your completly right! I've missed that :man_facepalming:

**Review by OttoWinter — APPROVED** [review]
Thanks and sorry for the long delay!

**Comment by abmantis** [comment]
@OttoWinter I actually still need to test this :/ That's why I left the WIP prefix. Sorry for the delay, but my dev ESP burned so I had to order another, which took some time.

**Comment by OttoWinter** [comment]
Ah ok, well the code looks good. And because it only expands the types it probably won't be a regression either, only maybe the feature doesn't work, but we can update that later too.

**Pull request merged**